spotbugs/src/main/java/edu/umd/cs/findbugs/ba/CheckReturnAnnotationDatabase.java

```diff
@@ -258,7 +273,68 @@ public CheckReturnValueAnnotation getResolvedAnnotation(Object o, boolean getMin
                 && ("java.lang.StringBuffer".equals(m.getClassName()) || "java.lang.StringBuilder".equals(m.getClassName()))) {
             return CheckReturnValueAnnotation.CHECK_RETURN_VALUE_MEDIUM;
         }
+        if (!AnalysisContext.currentAnalysisContext().isApplicationClass(m.getClassDescriptor())) {
+            // https://github.com/spotbugs/spotbugs/issues/429
+            // BuildCheckReturnAnnotationDatabase does not visit non-application classes,
+            // so we need to check package info dynamically
+
+            CheckReturnValueAnnotation packageDefault = packageInfoCache.computeIfAbsent(m.getPackageName(),
+                    this::parsePackage);
+            if (packageDefault != null) {
+                return packageDefault;
+            }
+        }
         return super.getResolvedAnnotation(o, getMinimal);
     }
 
+    private static final ClassDescriptor CHECK_RETURN_NULL_SPOTBUGS = DescriptorFactory
+            .createClassDescriptor(CheckReturnValue.class);
+    private static final ClassDescriptor CHECK_RETURN_NULL_JSR305 = DescriptorFactory
+            .createClassDescriptor("javax/annotation/CheckReturnValue");
+
+    private final Map<String, CheckReturnValueAnnotation> packageInfoCache = new HashMap<>();
+
+    /**
+     * Try to find default {@link CheckReturnValueAnnotation} for methods inside of target class.
+     *
+     */
+    @CheckForNull
+    private CheckReturnValueAnnotation parsePackage(@DottedClassName String packageName) {
+        String className = ClassName.toSlashedClassName(packageName) + "/package-info";
+        ClassDescriptor descriptor = DescriptorFactory.createClassDescriptor(className);
+        XClass clazz;
+        try {
+            clazz = descriptor.getXClass();
+        } catch (ResourceNotFoundException e) {
+            // no annotation on package
+            return null;
+        } catch (CheckedAnalysisException e) {
+            // ignore unexpected error to keep backward compatibility
+            e.printStackTrace();
```

Comment:
Can you please either don't print to std err at all or report analysis error? I think we should just join both catch blocks together.

Reply:
I don't agree, it will make debugging difficult and user cannot understand behavior in corner cases. 

How about printing stderr only when `FindBugs2.DEBUG` is set?

Reply:
OK.